packages/block-library/src/gallery/index.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Server-side rendering of the `core/image` block.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Handles backwards compatibility for Gallery Blocks,
+ * whose images feature a `data-id` attribute.
+ *
+ * Now that the Gallery Block contains inner Image Blocks,
+ * we add a custom `data-id` attribute before rendering the gallery
+ * so that the Image Block can pick it up in its render_callback.
+ *
+ * @param  array $parsed_block A single parsed block object.
+ *
+ * @return array               The migrated block object.
+ */
+function render_block_core_gallery_data( $parsed_block ) {
+	if ( 'core/gallery' === $parsed_block['blockName'] ) {
+		foreach ( $parsed_block['innerBlocks'] as $key => $inner_block ) {
+			if ( 'core/image' === $inner_block['blockName'] ) {
+				if ( ! $parsed_block['innerBlocks'][ $key ]['attrs']['data-id'] ) {
```

Comment:
It looks like this throws a PHP notice:

![image](https://user-images.githubusercontent.com/14988353/140452211-5f61c648-2408-4469-b6f1-66dd2eb27eb7.png)

Should this be an `isset()` check instead?

```suggestion
				if ( ! isset( $parsed_block['innerBlocks'][ $key ]['attrs']['data-id'] ) ) {
```

Reply:
Done